### ISSUE TITLE
jk transform - frontend command to modify configuration files

### ISSUE THREAD

**Issue by dlespiau** [jkcfg/jk#253]
`jk` can be used to load a file, modify it and write it back:

```
read('file.yaml').then(o => { o.metadata.name = 'foo'; write(o, 'result.yaml'); })
```

It could be nice to provide a dedicated `jk transform` command where the user only provides the script doing the modification. This way it could be used on a bunch of files we want to make the modification on:

```
jk transform -s add-istio-sidecar.js *-dep.yaml
jk transform -s set-ns.js -p ns=mynamespace *-dep.yaml
```

**Comment by squaremo** [comment]
UI sketch:

```
$ jk transform --help
Transform file contents with the given script

Usage:
    jk transform [flags] file... 

Flags:
  -c, --exec  treat value as literal JavaScript to run
  -m, --module resolve the given module name and use default export as script to run
  -o, --output-directory write transformed files to this directory, rather than overwriting them
  -p, --parameter name=value set input parameter
  -f, --parameters load parameters from a JSON or YAML file
  -s, --script the script to run, as a filesystem path
  --stdout print the transformed documents on stdout
  --stdin read the documents to be transformed from stdin
```

 - Having `-{c,m,s}` as mutually exclusive is slightly awkward, but consistent with `jk run` and `jk generate` (and mutually exclusive CLI flags are difficult anyway).
 - It's not clear what the default output should be. The safest is to print to stdout, but it might be a bit unexpected. Writing to the current directory would risk accidental overwrites (more so than `jk generate`, which doesn't expect a correspondence between the input files, if any, and the outputs).

**Comment by dlespiau** [comment]
It would be nice to have a `-i` flag for in place transformation.

Not too sure what the default should be. In place transform by default sounds like a recipe for disaster. Printing to stdout could work I guess, although if we have a mix of formats in the input files, it'll be awkward to do so.

:face_with_head_bandage:

**Comment by squaremo** [comment]
> It would be nice to have a -i flag for in place transformation.

As an explicit safeguard? I.e., it will refuse to write to the same files as its reading, unless you provide the `-i,--inplace` flag. That's a sound idea :+1: (I would probably drop the short form, or make it something else, since it collides with `-i, --input-directory`, which although not a flag here, might be guessed as such)

The `--stdout` option will either be for piping into something else, or just to eyeball the output. It's not a bad default, since it's harmless if you forget. Writing to `$PWD` (unless it would overwrite and `--inplace` was not passed) would be my second choice.

**Comment by squaremo** [comment]
Closed by #257
